activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -226,6 +226,23 @@ def ids
       pluck primary_key
     end
 
+    # Retrieve the records for the relation in raw format, _not_ typecasted.
+    # #select_rows returns an Array of Hashes representing the result set.
+    #
+    # This method is short-hand for
+    # <tt>ActiveRecord::Base.connection.exec_query(relation.to_sql).to_ary</tt>.
+    #
+    #   User.where(id: 1).select_rows
+    #   # => [{ "id" => 1, "name" => "Marco"}]
+    #
+    #   User.select('shard, count(*) as count').group(:shard).select_rows
+    #   # => [{ "shard" => 0, "count" => 16 }, { "shard" => 1, "count" => 17 }]
+    def select_rows
+      relation = spawn
+      result = skip_query_cache_if_necessary { klass.connection.select_all(relation.arel, nil) }
```

Comment:
I'm not sure if this `skip_query_cache_if_necessary` is necessary here. It's used by `#pluck`, where I got it from, but I'm not quite sure I understand why (or if) it's necessary for `#select_rows`.

Reply:
I think I _might_ have figured this caching situation out.

It seems like it comes from
 https://github.com/rails/rails/blob/068fe7dc9045856b822833db5cb7cb690e6000d7/activerecord/lib/active_record/relation/query_methods.rb#L897-L898
which is either called by the user, or by `relation.in_batches`.

With that in mind, it seems like we do need it here.

Reply:
skip_query_cache as far as I can remember was because using in_batches you don't want to cache the query results since you are going to perform a large scan and the cache doesn't make sense for that.